src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -72,66 +83,152 @@ public BalanceSheet getBalanceSheet(String transactionId, String companyAccounts
             throws ServiceException {
-
         ApiClient apiClient = apiClientService.getApiClient();
 
+        List<ValidationError> validationErrors = new ArrayList<>();
+
+        String smallFullUri = createUri(SMALL_FULL_URI, transactionId, companyAccountsId);
+        SmallFullApi smallFullApi = getSmallFullData(apiClient, smallFullUri);
+
+        CompanyProfileApi companyProfileApi = getCompanyProfile(companyNumber);
+        if (isMultipleYearFiler(companyProfileApi)) {
+            PreviousPeriodApi previousPeriodApi = transformer.getPreviousPeriod(balanceSheet);
+            String previousPeriodUri = createUri(PREVIOUS_PERIOD_URI, transactionId, companyAccountsId);
+            createPreviousPeriod(apiClient, smallFullApi, previousPeriodUri, previousPeriodApi, validationErrors);
+        }
+
         CurrentPeriodApi currentPeriod = transformer.getCurrentPeriod(balanceSheet);
+        String currentPeriodUri = CURRENT_PERIOD_URI.expand(transactionId, companyAccountsId).toString();
+        createCurrentPeriod(apiClient, smallFullApi, currentPeriodUri, currentPeriod, validationErrors);
 
-        String uri = CURRENT_PERIOD_URI.expand(transactionId, companyAccountsId).toString();
+        return validationErrors;
+
+    }
 
+    private SmallFullApi getSmallFullData(ApiClient apiClient, String smallFullUri)
+            throws ServiceException {
         try {
-            apiClient.smallFull().currentPeriod().create(uri, currentPeriod).execute();
+            return apiClient.smallFull().get(smallFullUri).execute();
         } catch (ApiErrorResponseException e) {
+            throw new ServiceException("Error retrieving small full data", e);
+        } catch (URIValidationException e) {
+            throw new ServiceException("Invalid URI for small full resource", e);
+        }
+    }
+
+    private void createPreviousPeriod(ApiClient apiClient, SmallFullApi smallFullApi, String previousPeriodUri, PreviousPeriodApi previousPeriodApi, List<ValidationError> validationErrors)
+            throws ServiceException {
+        boolean isCreated = isPreviousPeriodCreated(smallFullApi.getLinks());
+        try {
+            if (!isCreated) {
+                apiClient.smallFull().previousPeriod().create(previousPeriodUri, previousPeriodApi).execute();
+            } else {
+                apiClient.smallFull().previousPeriod().update(previousPeriodUri, previousPeriodApi).execute();
+            }
 
+        } catch (ApiErrorResponseException e) {
             if (e.getStatusCode() == HttpStatus.BAD_REQUEST.value()) {
-                List<ValidationError> validationErrors = validationContext.getValidationErrors(e);
-                if (validationErrors == null) {
+                validationErrors.addAll(validationContext.getValidationErrors(e));
+                if (validationErrors.isEmpty()) {
                     throw new ServiceException("Bad request posting balance sheet", e);
```

Comment:
As this exception is now thrown in the context of the `createPreviousPeriod()` method, it should be given a more specific exception message to differentiate from a similar exception thrown in the `createCurrentPeriod`() method, for example: `Bad request when creating previous period resource`.

Reply:
Since this exception can be thrown on the `POST` and `PUT` I will instead say, `Bad request when submitting previous period resource` as I feel `creating` gives the impression that it is just a `POST` and not `PUT`